firmware/quadruna/FSM/src/tasks.c

```diff
@@ -0,0 +1,130 @@
+#include "tasks.h"
+#include "main.h"
+#include "cmsis_os.h"
+
+#include "app_globals.h"
+#include "app_heartbeatMonitor.h"
+#include "states/app_driveState.h"
+
+#include "app_canTx.h"
+#include "app_canRx.h"
+#include "app_canAlerts.h"
+#include "app_commitInfo.h"
+
+#include "io_jsoncan.h"
+
+#include "hw_bootup.h"
+#include "hw_hardFaultHandler.h"
+#include "hw_watchdog.h"
+#include "hw_watchdogConfig.h"
+
+extern ADC_HandleTypeDef  *hadc1;
+extern CAN_HandleTypeDef  *hcan1;
+extern IWDG_HandleTypeDef *hiwdg;
+extern TIM_HandleTypeDef  *htim12;
+struct StateMachine       *state_machine;
+struct HeartbeatMonitor   *heartbeat_monitor;
+
+void canRxQueueOverflowCallBack(uint32_t overflow_count)
+{
+    app_canTx_FSM_RxOverflowCount_set(overflow_count);
+    app_canAlerts_FSM_Warning_TxOverflow_set(true);
+}
+
+void canTxQueueOverflowCallBack(uint32_t overflow_count)
+{
+    app_canTx_FSM_TxOverflowCount_set(overflow_count);
+    app_canAlerts_FSM_Warning_TxOverflow_set(true);
+}
+
+const CanConfig can_config = {
+    .rx_msg_filter        = NULL,
+    .tx_overflow_callback = canTxQueueOverflowCallBack,
+    .rx_overflow_callback = canRxQueueOverflowCallBack,
+};
+
+void tasks_preInit(void) {}
+
+void tasks_init(void)
+{
+    __HAL_DBGMCU_FREEZE_IWDG();
+
+    // Configure and initialize SEGGER SystemView.
+    SEGGER_SYSVIEW_Conf();
+
+    hw_hardFaultHandler_init();
+    hw_can_init(hcan1);
+    hw_watchdog_init(hw_watchdogConfig_refresh, hw_watchdogConfig_timeoutCallback);
+
+    __HAL_DBGMCU_FREEZE_IWDG();
```

Comment:
Looks like you got this line twice, not sure what that'll actually do but might as well delete.

Reply:
Done!